prow/sidecar/run.go

```diff
@@ -138,15 +159,29 @@ func (o Options) doUpload(spec *downwardapi.JobSpec, passed, aborted bool) error
 		result = "FAILURE"
 	}
 
+	// TODO(krzyzacy): Unify with downstream spyglass definition
 	finished := struct {
-		Timestamp int64  `json:"timestamp"`
-		Passed    bool   `json:"passed"`
-		Result    string `json:"result"`
+		Timestamp int64                  `json:"timestamp"`
+		Passed    bool                   `json:"passed"`
+		Result    string                 `json:"result"`
+		Metadata  map[string]interface{} `json:"metadata,omit_empty"`
+		Revision  string                 `json:"revision,omit_empty"`
 	}{
 		Timestamp: time.Now().Unix(),
 		Passed:    passed,
 		Result:    result,
+		Metadata:  metadata,
 	}
+
+	if spec.Refs != nil {
+		finished.Revision = spec.Refs.BaseSHA
+		if len(spec.Refs.Pulls) > 0 {
+			finished.Revision = spec.Refs.Pulls[0].SHA
+		}
+	} else if len(spec.ExtraRefs) > 0 {
+		finished.Revision = spec.ExtraRefs[0].BaseSHA
```

Comment:
I'm not sure that it make sense to handle `ExtraRefs` differently than `Refs`, that seems counter-intuitive.
Why wouldn't we check for the pull SHA in this case like we do for `Refs`?

Also note that `BaseSHA` isn't necessarily populated. Should we use `BaseRef` if the SHA is not specified?

Reply:
Is there anyone populating the Pulls in extraRefs?

Reply:
Not that I know of, but we can't just ignore the fields that we aren't using for our use cases. The field exists so we need to respect it if it is set.
What is the motivation for handling the `ExtraRefs[0]` in a different way than `Refs`? They both are `kube.Refs` so I would expect the `revision` to be determined in the same way.

Reply:
I'm updating that, 1min

Reply:
made a helper function